classes/class-dfrps-cpt.php

```diff
@@ -349,6 +359,141 @@ function cpt_dashboard_metabox( $post, $box ) {
+			echo '<option value="' . $value . '" ' . selected($i, $hour, false ) . '>' . $value . '</option>';
+        }
+		echo '</select>';
+		echo ":";
+		echo '<select name="minute">';
+		for ( $i = 0; $i <= 59; $i++ ){
+		    $value = ( 10 <= $i ) ? strval( $i) : '0' . $i;
+			echo '<option value="' . $value . '" ' . selected($i, $minute, false ) . '>' . $value . '</option>';
+		}
+		echo '</select>';
+		echo '</div>';
+
+		echo '</div>';
+    }
+
+	/**
+     * Build HTML for the "Day of week" multi select.
+	 * @param array $settings
+	 * @return string
+	 */
+    private function day_of_week( array $settings ) : string {
+        $html = '';
+		$show = 'Day of week' === $settings['interval'] ? '' : 'style="display:none"';
+		$html .= '<div class="section week" ' . $show . ' >';
+		$html .= '<label for="week">Days <span class="req">*</span></label>';
+		$html .= '<select name="week" id="week" class="day" multiple>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '1', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="1" ' . $selected . '>Every Monday</option>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '2', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="2" ' . $selected . '>Every Tuesday</option>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '3', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="3" ' . $selected . '>Every Wednesday</option>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '4', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="4" ' . $selected . '>Every Thursday</option>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '5', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="5" ' . $selected . '>Every Friday</option>';
+
+		$selected = 'Day of week' === $settings['interval'] && in_array( '6', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="6" ' . $selected . '>Every Saturday</option>';
+
+		$selected = in_array( '0', $settings['days'] ) ? 'selected' : '';
+		$html .= '<option value="0" ' . $selected . '>Every Sunday</option>';
+		$html .= '</select>';
+		$html .= '</div>';
+
+        return $html;
+    }
+
+	/**
+     * Build HTML for the "Day of the month" multi select.
+     *
+	 * @param $settings array
+	 * @return string
+	 */
+    private function day_of_month( array $settings ) : string {
+	    $html = '';
+
+		$show = 'Day of the month' === $settings['interval'] ? '' : 'style="display:none"';
+		$html .= '<div class="section month" ' . $show . '>';
+		$html .= '<label for="month">Days <span class="req">*</span></label>';
+		$html .= '<select name="month" id="month" class="day" multiple>';
+
+	    for ( $i=1; $i < 29; $i++ ) {
+            $selected = 'Day of the month' === $settings['interval'] && in_array( $i, $settings['days'] ) ? 'selected' : '';
+            switch ( $i ){
```

Comment:
Instead of a hardcoded switch statement here we could rely on `date_i18n()` which has the added benefit of translating the suffix into the user's local language.

```php
for ( $i = 1; $i < 29; $i ++ ) {
    $selected = 'Day of the month' === $settings['interval'] && in_array( $i, $settings['days'] ) ? 'selected' : '';
    $date     = sprintf( 'January %d, 2000', $i );
    
    $html .= '<option value="' . $i . '" ' . $selected . '>Every ' . esc_html( date_i18n( 'jS', strtotime( $date ) ) ) . '</option>';
}
```

Reply:
Added, thanks!